contracts/src/BeefyClient.sol

```diff
@@ -200,40 +200,24 @@ contract BeefyClient {
         nextValidatorSet = _nextValidatorSet;
     }
 
-    /* Public Functions */
+    /* External Functions */
 
     /**
-     * @dev Begin submission of commitment that was signed by the current validator set
+     * @dev Begin submission of commitment
      * @param commitment contains the commitment signed by the validators
      * @param bitfield a bitfield claiming which validators have signed the commitment
      * @param proof a proof that a single validator from currentValidatorSet has signed the commitment
      */
     function submitInitial(Commitment calldata commitment, uint256[] calldata bitfield, ValidatorProof calldata proof)
         external
     {
-        doSubmitInitial(currentValidatorSet, commitment, bitfield, proof);
-    }
+        if (commitment.validatorSetID != currentValidatorSet.id && commitment.validatorSetID != nextValidatorSet.id) {
```

Comment:
I think we should somehow combine this check with the code below. Because its duplicated arithmetic and an extra storage load which costs extra gas.

```solidity
bool is_next_validator_set = commitment.validatorSetID == nextValidatorSet.id;
```

Reply:
There's a duplicate, redundant vset.id check on line 243.